### PR TITLE
DIS2-1048 timeout logging

### PR DESCRIPTION
**Pull request by johnhuangclever**
Add kayvee log routing by bundling the kvconfig into the binary. We use https://github.com/jteeuwen/go-bindata to turn the yaml file into a go file and then it will build with the rest of the application. I added the steps for go-bindata generation into the Makefile.

This allows us to have a separate gearcmd-specific log routing for worker timeouts. We can eventually move over all the global gearcmd log routing rules over to this kvconfig file too.

Local run:
```
{"_kvmeta":{"kv_language":"go","kv_version":"6.9.0","routes":[{"dimensions":["function"],"rule":"worker-timed-out","series":"gearcmd.worker_timed_out","stat_type":"counter","type":"alerts","value_field":"value"}],"team":"UNSET"},"function":"john","level":"info","source":"gearcmd","timeout":4000000000,"title":"worker-timed-out","type":"counter","value":1}
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ COMPRESSED_BUILDS := $(BUILDS:%=%.tar.gz)
 RELEASE_ARTIFACTS := $(COMPRESSED_BUILDS:build/%=release/%)
 .PHONY: test $(PKGS) clean release vendor
 
-$(eval $(call golang-version-check,1.9))
+$(eval $(call golang-version-check,1.8))
```
Comment (johnhuangclever) [inline Makefile]: accidentally bumped this prematurely last PR

File: circle.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 machine:
   post:
   - cd $HOME && git clone --depth 1 -v git@github.com:clever/ci-scripts.git && cd ci-scripts && git show --oneline -s
-  - $HOME/ci-scripts/circleci/golang-install 1.9
+  - $HOME/ci-scripts/circleci/golang-install 1.8
```
Comment (johnhuangclever) [inline circle.yml]: accidentally bumped this prematurely last PR

File: kvconfig.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+routes:
+  worker-timed-out:
+    matchers:
+      title: [ "worker-timed-out" ]
+    output:
+      type: "alerts"
+      series: "gearcmd.worker_timed_out"
```
Comment (amir-clever) [inline kvconfig.yml]: 💯  great that we can now set up per worker alerts!
Should we also unconditionally slack timeouts to some channel? #gearman-timeouts ?

**Review by vynmeister — APPROVED** [review]

File: config/config.go [hunk 4]
```diff
@@ -2,8 +2,17 @@ package config
...
 var (
 	Clock clock.Clock = clock.New()
 )
+
+func init() {
+	data, err := Asset("kvconfig.yml")
+	if err != nil {
+		panic(err)
+	}
+	logger.SetGlobalRoutingFromBytes(data)
```
Comment (xavi-) [inline config/config.go]: Don't forget to add error check here.  `logger.SetGlobalRoutingFromBytes` returns an error.

File: config/kvconfig.go [hunk 5]
```diff
@@ -0,0 +1,235 @@
+// Code generated by go-bindata.
```
Comment (xavi-) [inline config/kvconfig.go]: Does this file need to build committed?  From what I understand it gets autogenerated when `make build` and `make release` are called.

**Review by xavi- — APPROVED** [review]
Looks good.  Left two comments.  Thanks for doing this!

File: config/kvconfig.go (hunk 5 above)
Comment (johnhuangclever) [reply config/kvconfig.go]: Yeah, unfortunately since the config.go file calls `Asset` which exists in this file.

**Pull request merged**